include/octree.h

```diff
@@ -10,31 +10,34 @@ class Octree {
     public:
 
         // Octree constructor
-        Octree(std::array<float, 2>& xCoords, std::array<float, 2>& yCoords, std::array<float, 2>& zCoords);
+        Octree(std::array<float, 2>& xCoords, std::array<float, 2>& yCoords, std::array<float, 2>& zCoords, bool internal);
 
         // Member functions
-        void clear();
+        void clearOctree();
+        void updateCoords(std::array<float, 2>& newXCoords, std::array<float, 2>& newYCoords, std::array<float, 2>& newZCoords);
         void insert(std::shared_ptr<T> objPtr);
         void build(std::vector<std::shared_ptr<T>>& objPtrs);
 
         // Members
         std::vector<std::shared_ptr<T>> objPtrs;
         std::array<float, 3> centerOfMass;
-        float* totalMass;
+        float totalMass;
+        bool internal;
 
         // Dimensions of the current octant
         std::array<float, 2> xCoords;
         std::array<float, 2> yCoords;
         std::array<float, 2> zCoords;
 
         // Octree children --> 0-7 based on 2D convention in postive z, and then 2D convention in negative z, observing from above        
-        Octree<T>* child0;
-        Octree<T>* child1;
-        Octree<T>* child2;
-        Octree<T>* child3;
-        Octree<T>* child4;
-        Octree<T>* child5;
-        Octree<T>* child6;
-        Octree<T>* child7;
+        std::shared_ptr<Octree<T>> child0;
```

Comment:
do u think u could add like comments for each of the children that just say which octants they refer to? Might be helpful for understanding the logic in octree.cpp